cargo-miri/bin.rs

```diff
@@ -585,6 +614,34 @@ fn phase_cargo_rustc(args: env::Args) {
         // (Need to do this here as cargo moves that "binary" to a different place before running it.)
         info.store(&out_filename("", ".exe"));
 
+        // Rustdoc expects us to exit with an error code if the test is marked as `compile_fail`,
+        // so we need to run Miri with `MIRI_BE_RUSTC` for a check-only build.
```

Comment:
I don't understand the command here. What do the error code and `MIRI_BE_RUSTC` have to to with each other?

Reply:
Oh I think I get it... so the point is that we'd *want* to just write the JSON file and be done, but actually we cannot since that would always succeed but we need to detect syntax errors. Can you extend the comment a bit?
```suggestion
        // so just creating the JSON is not enough: we need to detect syntax errors, so we
        // run Miri with `MIRI_BE_RUSTC` for a check-only build.
```